python-checks/src/main/java/org/sonar/python/checks/hotspots/SecureCookieCheck.java

```diff
@@ -0,0 +1,167 @@
+    "django.http.HttpResponsePermanentRedirect.set_cookie",
+    "django.http.HttpResponsePermanentRedirect.set_signed_cookie",
+    "django.http.HttpResponseNotModified.set_cookie",
+    "django.http.HttpResponseNotModified.set_signed_cookie",
+    "django.http.HttpResponseBadRequest.set_cookie",
+    "django.http.HttpResponseBadRequest.set_signed_cookie",
+    "django.http.HttpResponseNotFound.set_cookie",
+    "django.http.HttpResponseNotFound.set_signed_cookie",
+    "django.http.HttpResponseForbidden.set_cookie",
+    "django.http.HttpResponseForbidden.set_signed_cookie",
+    "django.http.HttpResponseNotAllowed.set_cookie",
+    "django.http.HttpResponseNotAllowed.set_signed_cookie",
+    "django.http.HttpResponseGone.set_cookie",
+    "django.http.HttpResponseGone.set_signed_cookie",
+    "django.http.HttpResponseServerError.set_cookie",
+    "django.http.HttpResponseServerError.set_signed_cookie",
+    "flask.Response.set_cookie"
+  ));
+
+  private static final String SECURE = "secure";
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Kind.ASSIGNMENT_STMT, ctx -> {
+      AssignmentStatement assignment = (AssignmentStatement) ctx.syntaxNode();
+      getSubscriptionToCookies(assignment.lhsExpressions())
+        .forEach(sub -> {
+          if (isSettingSecureFlag(sub) && isFalsy(assignment.assignedValue())) {
+            ctx.addIssue(assignment, MESSAGE);
+          }
+        });
+    });
+
+    context.registerSyntaxNodeConsumer(Kind.CALL_EXPR, ctx -> {
+      CallExpression callExpression = (CallExpression) ctx.syntaxNode();
+      Symbol calleeSymbol = callExpression.calleeSymbol();
+      if (calleeSymbol != null && SET_COOKIE_METHODS.contains(calleeSymbol.fullyQualifiedName())) {
+        if (callExpression.arguments().stream().anyMatch(argument -> argument.is(Kind.UNPACKING_EXPR))) {
+          return;
+        }
+        RegularArgument secureArgument = getSecureArgument(callExpression.arguments());
+        if (secureArgument == null || isFalsy(secureArgument.expression())) {
+          ctx.addIssue(callExpression.callee(), MESSAGE);
+        }
+      }
+    });
+  }
+
+  @CheckForNull
+  private static RegularArgument getSecureArgument(List<Argument> arguments) {
+    return arguments.stream()
+      .map(RegularArgument.class::cast)
+      .filter(argument -> {
+        Name keywordArgument = argument.keywordArgument();
+        return keywordArgument != null && keywordArgument.name().equals(SECURE);
+      })
+      .findFirst().orElse(null);
+  }
+
+  private static Stream<SubscriptionExpression> getSubscriptionToCookies(List<ExpressionList> lhsExpressions) {
+    return lhsExpressions.stream()
+      .flatMap(expressionList -> expressionList.expressions().stream())
+      .filter(lhs -> {
+        if (lhs.is(Kind.SUBSCRIPTION)) {
+          SubscriptionExpression sub = (SubscriptionExpression) lhs;
+          Symbol objectSymbol = getObjectSymbol(sub.object());
```

Comment:
Just a thought: do you think it would actually be feasible to raise on cases like: 
`cookie['c1'] = 'value' # FN`
by checking whether the `cookie` symbol has another assignment usage with the same `c1` subscript as well as a `secure` one?
Maybe it's not worth doing as I believe this is not the usual way to set cookies and it might be more complex to do than it looks, but I prefer mentioning it, feel free to disregard this!

Reply:
That's a fair point. However I have the feeling that the value doesn't justify the extra complexity. I suggest to create a ticket to keep track of this FN.

Reply:
Ticket created https://jira.sonarsource.com/browse/SONARPY-577